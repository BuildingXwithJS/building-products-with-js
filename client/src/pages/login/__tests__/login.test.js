// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import LoginPage, {Login} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '0',
  login: 'test',
  registrationDate: new Date(2016, 1, 1, 1, 1, 1, 1),
};

test('# Login page wrapper', () => {
  const store = mockStore({auth: {user}});
  const wrapper = shallow(<LoginPage store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Login page', () => {
  const newLogin = 'test';
  const newPassword = '123';
  const token = 'asd';
  const navToHome = () => expect(true).toBeTruthy();
  const onLoginClick = ({login, password, remember}) => {
    expect(login).toBe(newLogin);
    expect(password).toBe(newPassword);
    expect(remember).toBeTruthy();
  };

  const component = (
    <Login
      token={token}
      navToHome={navToHome}
      onLoginClick={onLoginClick}
    />
  );

  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  // mount for testing
  const app = mount(component);
  // set new login, pass and remember
  app.find('#inputUsername').getDOMNode().value = newLogin;
  app.find('#inputPassword').getDOMNode().value = newPassword;
  app.find('#inputRemember').getDOMNode().checked = true;
  // click login button
  app.find('button').simulate('click');
});
