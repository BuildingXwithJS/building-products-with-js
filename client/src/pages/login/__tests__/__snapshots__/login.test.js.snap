// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Login page 1`] = `
<div
  className="jumbotron"
>
  <h2>
    Experts portal:
  </h2>
  <p>
    Please log in. Or 
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/register"
    >
      register
    </Link>
  </p>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputUsername"
      >
        Username:
      </label>
      <input
        className="form-control"
        id="inputUsername"
        placeholder="Username"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="inputPassword"
        placeholder="Password"
        type="password"
      />
    </div>
    <div
      className="checkbox"
    >
      <label
        htmlFor="inputRemember"
      >
        <input
          id="inputRemember"
          type="checkbox"
        />
         Remember me
      </label>
    </div>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;

exports[`# Login page wrapper 1`] = `
<Login
  navToHome={[Function]}
  onLoginClick={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
